src/ResourceManager/MachineLearning/Commands.MachineLearning.Test/ScenarioTests/WebServiceTests.ps1

```diff
@@ -0,0 +1,330 @@
+            Clean-WebService $otherResourceGroupName $otherGroupWebServiceName
+            Clean-ResourceGroup $otherResourceGroupName 
+        }
+    };
+
+    RunWebServicesTest $actualTest
+}
+
+<#
+.SYNOPSIS
+Base function for running web services tests
+#>
+function RunWebServicesTest([ScriptBlock] $testScript)
+{
+    # Setup
+    $rgName = Get-ResourceGroupName 
+    $location = Get-ProviderLocation "Microsoft.MachineLearning" "webServices"
+    $webServiceName = Get-WebServiceName
+    $storageAccountName = Get-TestStorageAccountName
+    $commitmentPlanName = Get-CommitmentPlanName
+    $cpApiVersion = Get-ProviderAPIVersion "Microsoft.MachineLearning" "commitmentPlans"
+    LogOutput "Using version $cpApiVersion of the CP RP APIs"
+
+    try
+    {
+        # Setup
+        LogOutput "Creating resource group: $rgName"    
+        $group = New-AzureRmResourceGroup -Name $rgName -Location $location        
+        LogOutput("Created resource group: $($group.ResourceId)")
+
+        LogOutput "Creating storage account: $storageAccountName"    
+        $storageAccount = Create-TestStorageAccount $rgName $location $storageAccountName        
+        LogOutput("Created storage account: $storageAccountName")
+
+        LogOutput "Creating commitment plan resource: $commitmentPlanName"
+        $cpSku = @{Name = 'PLAN_SKU_NAME'; Tier='PLAN_SKU_TIER'; Capacity=1}
+        $cpPlan = New-AzureRmResource -Location $location -ResourceType "Microsoft.MachineLearning/CommitmentPlans" -ResourceName $commitmentPlanName -ResourceGroupName $rgName -SkuObject $cpSku -Properties @{} -ApiVersion $cpApiVersion -Force     
+        LogOutput "Created commitment plan resource: $($cpPlan.ResourceId)" 
+
+        &$testScript $rgName $location $webServiceName $cpPlan.ResourceId $storageAccount
+    }
+    finally
+    {  
+        Clean-TestStorageAccount $rgName $storageAccountName
+        Clean-ResourceGroup $rgName        
+    }
+}
+
+function LoadWebServiceDefinitionForTest([string] $filePath, [string] $commitmentPlanId, [object] $storageAccount)
+{
+    $svcDefinition = Import-AzureRmMlWebService -FromFile $filePath
+    $svcDefinition.Properties.CommitmentPlan.Id = $commitmentPlanId
+    $svcDefinition.Properties.StorageAccount.Name = $storageAccount.Name
+    $svcDefinition.Properties.StorageAccount.Key = $storageAccount.Key
+
+    return $svcDefinition
+}
+
+function ValidateWebServiceResult([string] $rgName, [string] $webServiceName, [string] $location, [Microsoft.Azure.Management.MachineLearning.WebServices.Models.WebService] $svc)
+{
+    $subscriptionId = ((Get-AzureRmContext).Subscription).SubscriptionId        
+    $expectedServiceResourceId = "/subscriptions/$subscriptionId/resourceGroups/$rgName/providers/Microsoft.MachineLearning/webservices/$webServiceName"
+    LogOutput "Checking that the created webservice's resource id $($svc.Id) matches the expected value $expectedServiceResourceId"
```

Comment:
Please don't do this, this subverts the current logging of powershell scripts.  Use the standard powershell streams in this case the Verbose stream and you'll get to see this as part of test logs.